docs: Add changelog for bt_mesh_adv_unref fix

PavelVPV opened the pull request:
Adds a change log for bt_mesh_adv_unref fix.

PR with the fix: https://github.com/nrfconnect/sdk-nrf/pull/22221

NordicBuilder commented:
## CI Information

<sup>_To view the history of this post, clich the 'edited' button above_</sup>
Build number: 1
### Inputs:
#### Sources:
sdk-nrf: PR head: [061f6d44dd3d15f516f99a13b82a05d202998621](https://github.com/nrfconnect/sdk-nrf/pull/22306/commits/061f6d44dd3d15f516f99a13b82a05d202998621)
<details><summary> more details</summary>

sdk-nrf:
> PR head: [061f6d44dd3d15f516f99a13b82a05d202998621](https://github.com/nrfconnect/sdk-nrf/pull/22306/commits/061f6d44dd3d15f516f99a13b82a05d202998621)
> merge base: [938fdd739c99ef5cefac6a479940d97112be668d](https://github.com/nrfconnect/sdk-nrf/commit/938fdd739c99ef5cefac6a479940d97112be668d)
> target head (v2.9-branch): [c64c882946ced3b3e2600a8821a0ac26d03b6897](https://github.com/nrfconnect/sdk-nrf/commit/c64c882946ced3b3e2600a8821a0ac26d03b6897)
> [Diff](https://github.com/nrfconnect/sdk-nrf/compare/938fdd739c99ef5cefac6a479940d97112be668d..061f6d44dd3d15f516f99a13b82a05d202998621)</details>

#### Github labels
| Enabled | Name | Description |
| -- | -- | -- |
|| ci-disabled | _Disable the ci execution_ | 
|| ci-all-test | _Run all of ci, no test spec filtering will be done_ | 
|| ci-force-downstream | _Force execution of downstream even if twister fails_ | 
|| ci-run-twister | _Force run twister_ | 
|| ci-run-zephyr-twister | _Force run zephyr twister_ | 

<details><summary> List of changed files detected by CI (1)</summary>

```
doc
│  ├── nrf
│  │  ├── releases_and_maturity
│  │  │  ├── releases
│  │  │  │  │ release-notes-changelog.rst
```
</details>

### Outputs:

#### Toolchain
Version: 
Build docker image: 

#### Test Spec & Results: :white_check_mark: Success; :x: Failure; :orange_circle: Queued; :yellow_circle: Progress; :white_medium_square: Skipped; :warning: Quarantine
- :orange_circle: Toolchain
- :orange_circle: Build twister
- :orange_circle: Integration tests

<details><summary>Disabled integration tests</summary>

- 
   - desktop52_verification
   - doc-internal
   - test_ble_nrf_config
   - test-fw-nrfconnect-apps
   - test-fw-nrfconnect-ble_mesh
   - test-fw-nrfconnect-ble_samples
   - test-fw-nrfconnect-boot
   - test-fw-nrfconnect-chip
   - test-fw-nrfconnect-fem
   - test-fw-nrfconnect-nfc
   - test-fw-nrfconnect-nrf-iot_libmodem-nrf
   - test-fw-nrfconnect-nrf-iot_lwm2m
   - test-fw-nrfconnect-nrf-iot_mosh
   - test-fw-nrfconnect-nrf-iot_nrf_provisioning
   - test-fw-nrfconnect-nrf-iot_positioning
   - test-fw-nrfconnect-nrf-iot_samples
   - test-fw-nrfconnect-nrf-iot_serial_lte_modem
   - test-fw-nrfconnect-nrf-iot_thingy91
   - test-fw-nrfconnect-nrf-iot_zephyr_lwm2m
   - test-fw-nrfconnect-nrf_crypto
   - test-fw-nrfconnect-ps
   - test-fw-nrfconnect-rpc
   - test-fw-nrfconnect-rs
   - test-fw-nrfconnect-tfm
   - test-fw-nrfconnect-thread
   - test-fw-nrfconnect-zigbee
   - test-low-level
   - test-sdk-audio
   - test-sdk-dfu
   - test-sdk-find-my
   - test-sdk-mcuboot
   - test-sdk-pmic-samples
   - test-sdk-sidewalk
   - test-sdk-wifi
   - test-secdom-samples-public
</details>



*Note: This message is automatically posted and updated by the CI*

peknis commented on doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:
```diff
@@ -395,7 +395,9 @@ Binary libraries
-|no_changes_yet_note|
+* :ref:`bt_mesh` library:
+
+  * Fixed an issue where ``bt_mesh_adv_unref()`` function could assert when messaging to a proxy node.
```
```suggestion
  * Fixed an issue where :c:func:`bt_mesh_adv_unref` function could assert when messaging to a proxy node.
```

PavelVPV replied:
It won't work as this is not a public API.

PavelVPV replied:
But customers figured out that assertion happens in this function (dev zone case for the reference: https://devzone.nordicsemi.com/f/nordic-q-a/117593/assertion-fail-west_topdir-zephyr-subsys-bluetooth-mesh-adv-c). That is why I think it is important to write function name here.

peknis replied:
Okay, then leave it with double backticks, but add *the*.

PavelVPV replied:
Added!

divipillai commented:
> This should be added to update docs for 2.9.2 PR: #22215

Thought of adding this to the 2.9.2 release notes after this PR is merged. But, it would be better to add it now, and close this PR.
Add the entry here: https://github.com/nrfconnect/sdk-nrf/pull/22215/files#diff-1eafd60ac3cd99890a8bc27774117cae2f8372f2eaa58b0afb26bc3f7b3f3570R132. We could close this PR. Thanks!

PavelVPV commented:
> > This should be added to update docs for 2.9.2 PR: #22215
> 
> Thought of adding this to the 2.9.2 release notes after this PR is merged. But, it would be better to add it now, and close this PR. Added the entry here: https://github.com/nrfconnect/sdk-nrf/pull/22215/files#diff-1eafd60ac3cd99890a8bc27774117cae2f8372f2eaa58b0afb26bc3f7b3f3570R132. We could close this PR. Thanks!

Then I close this PR and will post a comment in the mentioned PR instead.

PavelVPV commented:
> > > > This should be added to update docs for 2.9.2 PR: #22215
> > > 
> > > 
> > > Thought of adding this to the 2.9.2 release notes after this PR is merged. But, it would be better to add it now, and close this PR. Added the entry here: https://github.com/nrfconnect/sdk-nrf/pull/22215/files#diff-1eafd60ac3cd99890a8bc27774117cae2f8372f2eaa58b0afb26bc3f7b3f3570R132. We could close this PR. Thanks!
> > 
> > 
> > Then I close this PR and will post a comment in the mentioned PR instead.
> 
> I have already added the entry here: https://github.com/nrfconnect/sdk-nrf/pull/22215/files#diff-1eafd60ac3cd99890a8bc27774117cae2f8372f2eaa58b0afb26bc3f7b3f3570R132. Please check.

I see, thanks! Everything is done.
